pytest_operator/plugin.py

```diff
@@ -0,0 +1,108 @@
+import os
+import pytest
+import asyncio
+from pathlib import Path
+from juju.model import Model as JujuModel
+from aioify import aioify
+from subprocess import run
+
+
+def pytest_addoption(parser):
+
+    parser.addoption("--controller", action="store", help="Juju controller to use")
+
+    parser.addoption("--model", action="store", help="Juju model to use")
+
+    parser.addoption(
+        "--cloud", action="store", help="Juju cloud to use"
+    )
+
+
+class ToolsObj:
+    """Utility class for accessing juju related tools"""
+
+    def __init__(self, request):
+        self._request = request
+
+    async def _load(self):
+        request = self._request
+        self.controller_name = request.config.getoption("--controller")
+        self.model_name = request.config.getoption("--model")
+        self.cloud = request.config.getoption("--cloud")
+        self.connection = None
+        if self.model_name and self.cloud:
+            self.connection = f"{self.controller_name}:{self.model_name}"
+
+    async def run(self, cmd, *args, input=None):
```

Comment:
Why is this defined here and subprocess used later? If subprocess can simply be used with an async wrapper would that be better than writting our own run here?